src/parse.rs

```diff
@@ -2543,22 +2550,29 @@ impl<'a> Parser<'a> {
     }
 
     /// Returns the next byte offset on success.
-    fn scan_inline_html(&mut self, bytes: &[u8], ix: usize) -> Option<usize> {
+    fn scan_inline_html(&mut self, bytes: &[u8], ix: usize) -> Option<(Option<Vec<u8>>, usize)> {
```

Comment:
The comment above the function is now stale, and this is a case where it would probably be useful to document the type because it's not obvious.

Also, and I'm typing as I go, the `Option<Vec>` feels like it might not be necessary. Is `Some(vec![])` a valid response, important to be distinguished from `None`? If not, maybe just use `is_empty` for its consumer?

Reply:
Good insights. It turns out that `Option<Vec<u8>>` is not necessary indeed.